pinot-common/src/main/java/org/apache/pinot/common/request/context/LiteralContext.java

```diff
@@ -43,154 +39,202 @@
  */
 public class LiteralContext {
   // TODO: Support all of the types for sql.
-  private final FieldSpec.DataType _type;
+  private final DataType _type;
   private final Object _value;
-  private final BigDecimal _bigDecimalValue;
 
-  private static BigDecimal getBigDecimalValue(FieldSpec.DataType type, Object value) {
-    switch (type) {
-      case BIG_DECIMAL:
-        return (BigDecimal) value;
-      case BOOLEAN:
-        return PinotDataType.BOOLEAN.toBigDecimal(value);
-      case TIMESTAMP:
-        return PinotDataType.TIMESTAMP.toBigDecimal(Timestamp.valueOf(value.toString()));
-      default:
-        if (type.isNumeric()) {
-          return new BigDecimal(value.toString());
-        }
-        return BigDecimal.ZERO;
-    }
-  }
+  // PinotDataType is used for the type conversion.
+  private final transient PinotDataType _pinotDataType;
```

Comment:
why `transient`? This class is not serializable, right?

Reply:
Correct. Added `transient` for documentation purpose and future proof because it is not included in `equals()` and `hashCode()`. Certain tools such as `EqualsVerifier` won't pass if a non-transient field is not used.